.github/workflows/build-macos.yml

```diff
@@ -0,0 +1,81 @@
+---
+# CI code for build and test on macOS
+name: macOS Build and test
+on:
+  push: # Master branch checks only validate the build and generate artifacts for testing.
+    branches:
+      - master
+  pull_request: null # PR checks only validate the build and generate artifacts for testing.
+  workflow_dispatch: # Dispatch runs build and validate, then push to the appropriate storage location.
+    inputs:          # Inputs for this workflow can be omitted
+      type:
+        description: Build Type
+        default: nightly
+        required: true
+      version:
+        description: Version Tag
+        default: nightly
+        required: true
```

Comment:
This should be dropped, we have no need to run this on workflow_dispatch triggers.